scripts/check

```diff
@@ -5,10 +5,11 @@ if [ -d 'venv' ] ; then
     export PREFIX="venv/bin/"
 fi
 export SOURCE_FILES="httpcore tests"
+SOURCE_FILES_WITHOUT_SYNC=$(find httpcore tests -maxdepth 1 -not -path "*/_sync*" -type d)
 
 set -x
 
 ${PREFIX}ruff check --show-source $SOURCE_FILES
-${PREFIX}black --exclude '/(_sync|sync_tests)/' --check --diff $SOURCE_FILES
+${PREFIX}ruff format $SOURCE_FILES_WITHOUT_SYNC --diff
```

Comment:
The `SOURCE_FILES_WITHOUT_SYNC=...` is a bit opaque.
Can we use a less complex approach?

Reply:
I believe we can entirely remove it.
Ruff will use the "excludes" setting from the `pyproject.toml` file.

Demo:

``` shell
ruff httpcore/ tests --verbose

[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Ignored path via `exclude`: "/home/test/programs/gitprojects/httpcore/tests/_sync"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_synchronization.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_exceptions.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Ignored path via `exclude`: "/home/test/programs/gitprojects/httpcore/httpcore/_sync"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_utils.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/tests/test_models.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_api.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_backends/__init__.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/_backends/trio.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/tests/test_api.py"
[2023-11-13][11:03:56][ruff_workspace::resolver][DEBUG] Included path via `include`: "/home/test/programs/gitprojects/httpcore/httpcore/__init__.py"
```